### DISCUSSION TITLE
relation of otel/spring-boot-autoconfigure and spring/spring-cloud-sleuth-otel

### DISCUSSION THREAD

**Discussion by warmuuh** [open-telemetry/opentelemetry-java-instrumentation#2835] [Q&A] [answered]
I am currently diving into how to instrument spring code with otel and i saw that there are two alternative spring-boot autoconfigurations, one by otel and one in the spring-incubator (the spring-incubator one (spring-cloud-sleuth-otel) was apparently created more recently)

what is the idea around this? will the spring-maintained version replace the one by otel? and how far will this impact the spring-specific instrumentation made by the java-agent? (as it is kind of duplicate work)

**Comment by trask** [comment, marked as answer]
hi @warmuuh!

> will the spring-maintained version replace the one by otel?

hopefully yes

> and how far will this impact the spring-specific instrumentation made by the java-agent?

the goal is for the java agent to re-use (and inject) library instrumentation where possible, to avoid duplication and ensure consistent coverage/behavior, but it varies a bit from instrumentation to instrumentation
